src/celestia/qt/qtappwin.cpp

```diff
@@ -136,6 +136,47 @@ class AppAlerter : public CelestiaCore::Alerter
     QWidget* parent;
 };
 
+class FPSActionGroup
+{
+    QActionGroup *m_group;
+    std::map<int, QAction*> m_map;
+    QAction *m_custom;
+    int m_last;
+public:
+    FPSActionGroup(QObject *p = nullptr)
+    {
+        QAction *fps;
+        std::array<int, 5> fps_array = { 0, 15, 30, 60, 120 };
+
+        m_group = new QActionGroup(p);
+        for (auto ifps : fps_array)
+        {
+            fps = new QAction(ifps == 0 ? _("Auto") : QString::number(ifps), p);
+            fps->setCheckable(true);
+            fps->setData(ifps);
+            m_map[ifps] = fps;
+            m_group->addAction(fps);
+    }
+        m_custom = new QAction(_("Custom"), p);
+        m_custom->setCheckable(true);
+        m_custom->setShortcut(QString("Shift+F"));
+        m_group->addAction(m_custom);
+        m_group->setExclusive(true);
+    }
+
+    std::map<int, QAction*> &actions() { return m_map; }
+    QAction *custom() { return m_custom; }
+    int last() { return m_last; }
+    void update(int fps)
```

Comment:
This method's implementation shouldn't be inside the class definition because it will be inlined.

Reply:
Indeed, however custom() and last() should be rather inline anyway.